bootstrap/config/config.go

```diff
@@ -1271,3 +1280,76 @@ func GetInsecureSecretDataFullPath(secretName, key string) string {
 	return fmt.Sprintf("%s/%s/%s/%s/%s",
 		writableKey, insecureSecretsKey, secretName, secretDataKey, key)
 }
+
+func PutMissingConfigToConfigProvider(yamlFilePath string, secretProvider interfaces.SecretProvider, configClient configuration.Client, lc logger.LoggingClient) error {
+	keyValues, err := KeyValuesFromYamlFile(yamlFilePath, secretProvider, lc)
+	if err != nil {
+		return fmt.Errorf("failed to put missing config by yaml file %s: %s", yamlFilePath, err.Error())
+	}
+
+	for key, value := range keyValues {
+		if value == nil {
+			continue
+		}
+		hasSubConfig, err := configClient.HasSubConfiguration(key)
+		if err != nil {
+			return fmt.Errorf("failed to check config %s: %s", key, err.Error())
+		}
+		if hasSubConfig {
+			continue
+		}
+		if err = configClient.PutConfigurationValue(key, []byte(fmt.Sprint(value))); err != nil {
+			return fmt.Errorf("failed to push common configuration key %s with value %v: %s", key, value, err.Error())
+		}
+	}
+	return nil
+}
+
+// KeyValuesFromYamlFile loads yaml file and convert to key value pairs
+//
+//	For example,
+//	{
+//		"app-services/Writable/StoreAndForward/MaxRetryCount": 10,  <= common config
+//		"Writable/MaxDevices": 0,  <= private config
+//	}
+func KeyValuesFromYamlFile(yamlFile string, secretProvider interfaces.SecretProvider, lc logger.LoggingClient) (map[string]any, error) {
+	contents, err := file.Load(yamlFile, secretProvider, lc)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read configuration file %s: %s", yamlFile, err.Error())
+	}
+
+	data := make(map[string]any)
+	kv := make(map[string]any)
+
+	err = yaml.Unmarshal(contents, &data)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshall yaml configuration: %s", err.Error())
+	}
+
+	kv = buildKeyValues(data, kv, "")
+	return kv, nil
+}
+
+// buildKeyValues is a helper function to parse the configuration yaml file contents
+func buildKeyValues(data map[string]interface{}, kv map[string]interface{}, origKey string) map[string]interface{} {
+	key := origKey
+	for k, v := range data {
+		if len(key) == 0 {
+			key = fmt.Sprint(k)
+		} else {
+			key = fmt.Sprintf("%s/%s", key, k)
+		}
+
+		vdata, ok := v.(map[string]interface{})
+		if !ok {
+			kv[key] = v
+			key = origKey
+			continue
+		}
+
+		kv = buildKeyValues(vdata, kv, key)
+		key = origKey
+	}
+
+	return kv
+}
```

Comment:
The current function comment doesn’t clearly convey the purpose of this helper, making it hard to judge whether its recursive logic is correct.

From the code, I can only guess that the function’s intent is to flatten a nested map into a single-level map? If that’s correct, please update the comment to explicitly describe this purpose. Also, since this is a utility function, consider to move this func to bootstrap/utils/utils.go and this func should have unit tests to verify that its behavior matches the intended goal.